src/Storefront/Theme/ThemeScripts.php

```diff
@@ -41,34 +38,31 @@ public function getThemeScripts(): array
             return [];
         }
 
-        $themeName = $request->attributes->get(SalesChannelRequest::ATTRIBUTE_THEME_NAME, SalesChannelRequest::ATTRIBUTE_THEME_BASE_NAME)
+        $themeName = $request->attributes->get(SalesChannelRequest::ATTRIBUTE_THEME_NAME)
             ?? $request->attributes->get(SalesChannelRequest::ATTRIBUTE_THEME_BASE_NAME);
 
         $themeId = $request->attributes->get(SalesChannelRequest::ATTRIBUTE_THEME_ID);
+        $salesChannelId = $request->attributes->get(PlatformRequest::ATTRIBUTE_SALES_CHANNEL_ID) ?? null;
 
-        if ($themeName === null || $themeId === null) {
+        if ($themeName === null || $themeId === null || $salesChannelId === null) {
             return [];
         }
 
-        $salesChannelId = $request->attributes->get(PlatformRequest::ATTRIBUTE_SALES_CHANNEL_ID);
-        $path = $this->themePathBuilder->assemblePath($salesChannelId, $themeId);
-
         $salesChannelContext = $request->attributes->get(PlatformRequest::ATTRIBUTE_SALES_CHANNEL_CONTEXT_OBJECT);
-
         if (!$salesChannelContext instanceof SalesChannelContext) {
             return [];
         }
 
-        return $this->cache->get('theme_scripts_' . $path, function (ItemInterface $item) use ($themeId, $salesChannelContext) {
-            $themeConfig = $this->configLoader->load($themeId, $salesChannelContext->getContext());
+        $path = $this->themePathBuilder->assemblePath($salesChannelId, $themeId);
+
+        return $this->cache->get('theme_scripts_' . $path, function (ItemInterface $item) use ($themeId) {
```

Comment:
do we really need this cache here anymore 🤔, when the query is simple maybe we can skip this